documentation/USAGE.rst

```diff
@@ -535,68 +535,223 @@ Further demonstrations of spatially varying boundary condition can be found
+
+.. math::
+
+   \begin{aligned}
+       \nabla\cdot\left(\vec{u}\phi\right) &\approx
+       \sum_f \left(\hat{n}\cdot\vec{u}\right)_f \phi_f A_f \\
+       &= \sum_{f\neq f_0} \left(\hat{n}\cdot\vec{u}\right)_f \phi_f A_f
+       + \left(\hat{n}\cdot\vec{u}\right)_{f_0} \frac{g - b \left(\hat{n}\cdot\nabla\phi\right)_{f_0}}{a} A_{f_0} \\
+       &= \sum_{f\neq f_0} \left(\hat{n}\cdot\vec{u}\right)_f \phi_f A_f
+       + \left(\hat{n}\cdot\vec{u}\right)_{f_0}
+            \frac{-g \left(\hat{n}\cdot\vec{d}_{fP}\right)_{f_0} + b\phi_P}
+                 {-a \left(\hat{n}\cdot\vec{d}_{fP}\right)_{f_0} + b} A_{f_0}
+   \end{aligned}
+
+Applying internal "boundary" conditions
+=======================================
 
 Applying internal boundary conditions can be achieved through the use
-of implicit and explicit sources. An equation of the form
+of implicit and explicit sources. 
+
+Internal fixed value
+--------------------
+
+An equation of the form
 
 >>> eqn = TransientTerm() == DiffusionTerm()
 
 can be constrained to have a fixed internal ``value`` at a position
 given by ``mask`` with the following alterations
 
->>> eqn = TransientTerm() == DiffusionTerm() - ImplicitSourceTerm(mask * largeValue) + mask * largeValue * value
+>>> eqn = (TransientTerm() == DiffusionTerm() 
+...                           - ImplicitSourceTerm(mask * largeValue) 
+...                           + mask * largeValue * value)
 
 The parameter ``largeValue`` must be chosen to be large enough to
 completely dominate the matrix diagonal and the RHS vector in cells
 that are masked. The ``mask`` variable would typically be a
 ``CellVariable`` boolean constructed using the cell center values.
 
-One must be careful to distinguish between constraining internal cell
-values during the solve step and simply applying arbitrary constraints
-to a ``CellVariable``. Applying a constraint,
+Internal fixed gradient
+-----------------------
+
+An equation of the form
+
+>>> eqn = TransientTerm() == DiffusionTerm(coeff=Gamma0)
+
+can be constrained to have a fixed internal ``gradient`` magnitude 
+at a position given by ``mask`` with the following alterations
+
+>>> Gamma = FaceVariable(mesh=mesh, value=Gamma0)
+>>> Gamma[mask.value] = 0.
+>>> eqn = (TransientTerm() == DiffusionTerm(coeff=Gamma) 
+...        + DiffusionTerm(coeff=largeValue * mask)
+...        - ImplicitSourceTerm(mask * largeValue * gradient 
+...                             * mesh.faceNormals).divergence)
+
+The parameter ``largeValue`` must be chosen to be large enough to
+completely dominate the matrix diagonal and the RHS vector in cells
+that are masked. The ``mask`` variable would typically be a
+``FaceVariable`` boolean constructed using the face center values.
+
+Internal Robin condition
+------------------------
+
+Nothing different needs to be done when 
+`applying Robin boundary conditions`_ at internal faces.
+
+.. note::
+
+   While we believe these derivations are "correct", they often do not 
```

Comment:
Which derivations and what doesn't work? I'm not sure what you're referring to. Do you mean that "While the derivations above work well for external boundary conditions they fail to work for internal boundary conditions"?

Reply:
The derivation for how to modify a DiffusionTerm and how to modify a ConvectionTerm. Oh, I see. I didn't mean that to be under "Internal Robin condition". It's a generic issue: I couldn't make that substitute-and-extrapolate approach work for `examples/convection/robin.py`, but I think the generic idea is on the right track for a Robin condition that isn't "just the flux".

Reply:
Clarified that the earlier Robin derivation is believed "correct"